MHVLib/HTTP/MHVHttpService.m

```diff
@@ -0,0 +1,276 @@
+//
+// MHVHttpService.m
+// MHVLib
+//
+//  Copyright (c) 2017 Microsoft Corporation. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+#import "MHVHttpService.h"
+#import "MHVHttpServiceResponse.h"
+#import "Logger.h"
+#import "MHVBlobSource.h"
+#import "MHVValidator.h"
+
+@interface MHVHttpService () <NSURLSessionDelegate>
+
+@property (nonatomic, strong) NSURLSession *urlSession;
+@property (nonatomic, strong) NSOperationQueue *certificateCheckQueue;
+
+@property (nonatomic, assign) NSInteger requestCount;
+
+@end
+
+@implementation MHVHttpService
+
+- (instancetype)init
+{
+    self = [super init];
+    if (self)
+    {
+        _urlSession = [NSURLSession sessionWithConfiguration:[NSURLSessionConfiguration defaultSessionConfiguration]
```

Comment:
Added initializer so NSURLSession can be passed in, and init that creates a default (NSURLSession has a delegate that can only be set when it's created)